### DISCUSSION TITLE
exclude_polygons limitations

### DISCUSSION THREAD

**Discussion by canbilgen** [valhalla/valhalla#4754] [Q&A] [answered]
Hi, when using exclude_polygons while calculating a route using the demo app(valhalla.openstreetmap.de) sometimes I get an error stating "OSM: Exceeded maximum circumference for exclude_polygons: 100000 meters". Is there a setting which I can change or this is a hard limit in the  valhalla API?
Thanks.

**Comment by chrstnbwnkl** [comment, marked as answer]
It's a server side setting, so you can't control it with a request parameter. The settings path in the valhalla config is `.service_limits.max_exclude_polygons_length`. The reason this value is so conservative is that the current implementation of polygon-based exclusion of edges leads to high memory use, as all edge IDs inside the user provided polygon are collected into an in memory set used for lookups during the expansion. There's a discussion though to change the way its implemented: https://github.com/valhalla/valhalla/issues/4387

**Reply by canbilgen** [reply]
Thanks for fast response, I will change it in the valhalla.json but is there a maximum value limit?

**Reply by nilsnolde** [reply]
no theoretical one, but a practical one: it becomes super slow. if you'd try to avoid a whole country this way you'd wait for many minutes I think. the issue @chrstnbwnkl is referencing lays out a plan to even make country-sized polygons kinda work, at least exponentially faster than currently.

**Reply by canbilgen** [reply]
I appreciate the help you guys provided, thank you so much.
